src/vm/jithelpers.cpp

```diff
@@ -456,7 +456,7 @@ HCIMPLEND
 HCIMPL2_VV(UINT64, JIT_LLsh, UINT64 num, int shift)
 {
     FCALL_CONTRACT;
-    return num << shift;
+    return num << (shift & 0x3F);
```

Comment:
So the general plan is to have everything (except the actual ARM32 shift instructions, of course) mask the shift count, right?

Reply:
Yes. I think that's the best way to ensure that all various compilation modes and architectures give back the same result, and it mirrors what C# does.